goldilocks/src/lib.rs

```diff
@@ -321,3 +321,86 @@ unsafe fn add_no_canonicalize_trashing_input(x: u64, y: u64) -> u64 {
     // Below cannot overflow unless the assumption if x + y < 2**64 + ORDER is incorrect.
     res_wrapped + Goldilocks::NEG_ORDER * u64::from(carry)
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    type F = Goldilocks;
+
+    #[test]
+    fn test_goldilocks() {
+        let f = F::new(100);
+        assert_eq!(f.as_canonical_u64(), 100);
+
+        let f = F::new(u64::MAX);
```

Comment:
Maybe a quick comment about the reduction that makes these two equal?

Reply:
Added a short comment on it